cmd/server/main.go

```diff
@@ -100,3 +206,134 @@ func rootHandler(s storage.Storage) http.HandlerFunc {
 		w.Write([]byte("</ul></body></html>"))
 	}
 }
+
+// updateJSONHandler handles POST /update/ with JSON body
+func updateJSONHandler(s storage.Storage) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Content-Type") != "application/json" {
+			http.Error(w, "Content-Type must be application/json", http.StatusBadRequest)
+			return
+		}
+
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, "Failed to read request body", http.StatusBadRequest)
+			return
+		}
+
+		var metric models.Metrics
+		if err := json.Unmarshal(body, &metric); err != nil {
+			http.Error(w, "Invalid JSON", http.StatusBadRequest)
+			return
+		}
+
+		// Validate required fields
+		if metric.ID == "" || metric.MType == "" {
+			http.Error(w, "ID and MType are required", http.StatusBadRequest)
+			return
+		}
+
+		switch metric.MType {
+		case GaugeType:
+			if metric.Value == nil {
+				http.Error(w, "Value is required for gauge metrics", http.StatusBadRequest)
+				return
+			}
+			s.UpdateGauge(metric.ID, *metric.Value)
+			// Return the updated metric
+			response := models.Metrics{
+				ID:    metric.ID,
+				MType: metric.MType,
+				Value: metric.Value,
+			}
+			w.Header().Set("Content-Type", "application/json")
+			json.NewEncoder(w).Encode(response)
+
+		case CounterType:
+			if metric.Delta == nil {
+				http.Error(w, "Delta is required for counter metrics", http.StatusBadRequest)
+				return
+			}
+			s.UpdateCounter(metric.ID, *metric.Delta)
+			// Get the updated value from storage
+			if updatedValue, ok := s.GetCounter(metric.ID); ok {
+				response := models.Metrics{
+					ID:    metric.ID,
+					MType: metric.MType,
+					Delta: &updatedValue,
+				}
+				w.Header().Set("Content-Type", "application/json")
+				json.NewEncoder(w).Encode(response)
+			} else {
+				http.Error(w, "Failed to retrieve updated counter value", http.StatusInternalServerError)
+				return
+			}
```

Comment:
Мне кажется стоит проверить негативный кейс и сразу выйти.
```suggestion
			if updatedValue, ok := s.GetCounter(metric.ID); !ok {
			        http.Error(w, "Failed to retrieve updated counter value", http.StatusInternalServerError)
				return
			}
		        response := models.Metrics{
			        ID:    metric.ID,
			        MType: metric.MType,
			        Delta: &updatedValue,
		        }
		        w.Header().Set("Content-Type", "application/json")
		        json.NewEncoder(w).Encode(response)
```